nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,799 @@
+"""High-level decoding object that exposes standard classification
+
+and regression strategies such as SVM, LogisticRegression and Ridge,
+with optional feature selection, and integrated parameter selection.
+"""
+# Authors : Yannick Schwartz
+#           Andres Hoyos-Idrobo
+#
+# License: simplified BSD
+
+import sklearn
+import itertools
+import warnings
+import numpy as np
+from distutils.version import LooseVersion
+from sklearn.externals.joblib import Parallel, delayed, Memory
+from sklearn.linear_model.ridge import Ridge, RidgeClassifier, _BaseRidge
+from sklearn.linear_model import LogisticRegression
+from sklearn.svm import LinearSVC, SVR
+from sklearn.preprocessing import LabelBinarizer
+from sklearn.svm.bounds import l1_min_c
+from sklearn.base import RegressorMixin
+from sklearn.linear_model.base import LinearModel
+from sklearn.utils.extmath import safe_sparse_dot
+from sklearn import clone
+
+try:
+    from sklearn.utils import atleast2d_or_csr
+except ImportError: # sklearn 0.15
+    from sklearn.utils import check_array as atleast2d_or_csr
+
+try:
+    from sklearn.grid_search import ParameterGrid
+except ImportError: # sklearn 0.18
+    from sklearn.model_selection import ParameterGrid
+
+from ..input_data.masker_validation import check_embedded_nifti_masker
+from .._utils.param_validation import _adjust_screening_percentile
+from .._utils.fixes import check_scoring
+from .._utils.fixes import check_X_y
+from .._utils.fixes import check_is_fitted
+from .._utils.compat import _basestring
+from .._utils.fixes import check_cv
+from .._utils.param_validation import check_feature_screening
+from .._utils import CacheMixin
+
+
+SUPPORTED_ESTIMATORS = dict(
+    svc_l1=LinearSVC(penalty='l1', dual=False),
+    svc_l2=LinearSVC(penalty='l2'),
+    svc=LinearSVC(penalty='l2'),
```

Comment:
Thats because you know scikit learn API very well. Not necessary that everybody knows scikit learn API. For instance, neuroimaging user.

Atleast you should try state that in documentation